src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecutionTest.java

```diff
@@ -241,6 +242,112 @@ private static void assertStepExecutions(FlowExecution e, String... steps) throw
         });
     }
 
+    @Issue("JENKINS-34256")
+    @Test public void quietDownThenCancelQuietDown() {
+        story.then(r -> {
+            WorkflowJob p = r.createProject(WorkflowJob.class);
+            p.setDefinition(new CpsFlowDefinition("semaphore 'wait'; echo 'I am done'", true));
+            WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+            SemaphoreStep.waitForStart("wait/1", b);
+            r.jenkins.doQuietDown(true, 0);
+            SemaphoreStep.success("wait/1", null);
+            r.waitForMessage("Pausing (Preparing for shutdown)", b);
+            assertTrue(b.isBuilding());
+            r.assertLogNotContains("I am done", b);
+            r.jenkins.doCancelQuietDown();
+            r.waitForMessage("Resuming (Prepare for shutdown was canceled)", b);
+            r.assertLogContains("I am done", r.assertBuildStatusSuccess(r.waitForCompletion(b)));
+        });
+    }
+
+    @Issue("JENKINS-34256")
+    @Test public void pauseThenQuietDownThenUnpauseThenCancelQuietDown() {
```

Comment:
I'm not sure about all of these tests; I think they might be flaky. I added them because I was trying to make sure there weren't any bugs around combinations of pause and quiet mode. Some of the cases are a bit weird, especially when you unpause or cancel quiet mode when quiet mode is enabled or when the build is paused, respectively. I think I could tweak `CpsFlowExecution.pause()` to avoid printing consecutive "Pausing" and "Resuming" lines, but I'm not sure if that's worth the complexity.

Reply:
> I'm not sure if that's worth the complexity

Likely not.

Reply:
Tests seem good when I run them. I see the need for all the combinations, but yeah it does  kind of suck to have  to do all these variations. There's probably a parameterized way to simplify the code, but I don't think it's worth the effort. Maybe put it another test class?

Reply:
Yeah, using a parameterized test would probably make the intention clearer, but I think the abstraction might make the individual tests more difficult to understand, so I'd prefer to leave them as-is.